Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,162 @@
+===========
+Virtual Bus
+===========
+
+See <linux/virtual_bus.h> for the models for virtbus_device and virtbus_driver.
+
+In some subsystems, the functionality of the core device (PCI/ACPI/other) may
+be too complex for a single device to be managed as a monolithic block.
+Splitting the functionality into smaller orthogonal devices would make it
+easier to manage data, power management and domain-specific communication with
```

Comment:
Curious. Agree that it makes domain specific communication easier. Do we believe it makes PM easier too?

Reply:
In the case of audio with separate cards for HDMI and local mics/cpeakers/DMIC for example, splitting them up into 2 devices would let the PM manage only the devices within each card. So fro example, starting playback on HDMI, would only set up the components required for the HDMI pipelines where today since they are all part of the same device, we set up all the components in all the pcm devices.